ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt

```diff
@@ -26,27 +26,53 @@ internal sealed class OfferingSelection {
 class PaywallOptions(builder: Builder) {
 
     internal val offeringSelection: OfferingSelection
-    private val shouldDisplayDismissButton: Boolean
+    internal val shouldDisplayDismissButton: Boolean
     val fontProvider: FontProvider?
     val listener: PaywallListener?
-    internal var mode: PaywallMode = PaywallMode.default
+    internal val mode: PaywallMode
     val dismissRequest: () -> Unit
 
     init {
         this.offeringSelection = builder.offeringSelection
         this.shouldDisplayDismissButton = builder.shouldDisplayDismissButton
         this.fontProvider = builder.fontProvider
         this.listener = builder.listener
+        this.mode = builder.mode
         this.dismissRequest = builder.dismissRequest
     }
 
+    @Suppress("LongParameterList")
+    internal fun copy(
```

Comment:
In order to ignore whether to display the close button parameter in the footer, I had to do something like this... I don't fully like that this has the chance of us forgetting to add any new parameters in this copy method... But I think it might be ok.

Reply:
Yeah, we could use data classes for the options instead and we would get the copy function. However, in that case there is no way to make the visibility of some of the extra methods added in data classes internal... It might be ok but it's not recommended to expose data classes as part of the public API if possible, and wanted to follow the same pattern we have with other builder classes